extensions/functions_arithmetic.yaml

```
@@ -1071,6 +1071,125 @@ aggregate_functions:
             value: fp64
         nullability: DECLARED_OUTPUT
         return: fp64?
+  - name: "median"
+    description: > 
+      Calculate the median for a set of values.
+
+      Returns null if applied to zero records. For the integer implementations,
+      the rounding option determines how the median should be rounded if it ends
+      up midway between two values. For the floating point implementations,
+      they specify the usual floating point rounding mode.
+    impls:
+      - args:
+          - name: precision
+            description: >
+              Based on required operator performance and configured optimizations
+              on saving memory bandwidth, the precision of the end result can be
+              the highest possible accuracy of an approximation.
+
+                - EXACT: provides the highest accurate output
+                - APPROXIMATE: provides a sub-optimal output
```

This still doesn't specify *how* approximate the result may be. Is this what you mean, or is this too broad?

```suggestion
                - APPROXIMATE: provides only an estimate; the result must lie
                  between the minimum and maximum values in the input
                  (inclusive), but otherwise the accuracy is left up to the
                  consumer
```

---

I wanted to mean this one, but may be we could elaborate this a bit? 

Although, this is broad since the optimization strategy is very hard to specify. Again I think the choice of approximation is up to the engine to decide and that depends on various optimization techniques. But that's not the job of Substrait to define that optimization strategy. Substrait can only specify that it is going to be an approximation. So I wanted to put forward that idea. 

Should we enhance more?

---

I don't know that we can do better than what I suggested to constrain the approximation method further, but you're welcome to try. I at least can't think of any approximate median algorithm that doesn't at least satisfy that constraint, and only trivial constraints is better than no constraints, IMO.

---

Let's keep the suggested once for now. I will update this.